driver/src/test/java/org/neo4j/driver/internal/cluster/RediscoveryTest.java

```diff
@@ -454,10 +455,22 @@ private static ClusterComposition rediscover( BoltServerAddress initialRouter, C
         Clock mockedClock = mock( Clock.class );
         Logger mockedLogger = mock( Logger.class );
 
-        Rediscovery rediscovery = new Rediscovery( initialRouter, settings, mockedClock, mockedLogger, provider );
+        Rediscovery rediscovery = new Rediscovery( initialRouter, settings, mockedClock, mockedLogger, provider,
+                directMapProvider );
         return rediscovery.lookupClusterComposition( connections, routingTable );
     }
 
+    private static DnsResolver directMapProvider = new DnsResolver()
```

Comment:
Maybe introduce a `private static class` for this or move it to the top of the class? It is just a bit strange to see field in the middle of the class.